### DISCUSSION TITLE
Relative shadow blur sizes

### DISCUSSION THREAD

**Discussion by lostminds** [godotengine/godot-proposals#6117] [3D]
I'm posting this here as I'm not sure if this is a bug or a missing feature that I'd like to see. 

Basically the issue I've run into is that when making soft shadows the blur size property (at least on directional lights) seems to be absolute, related to some pixel blur size. The issue with this is that if you change the shadow atlas size (for performance settings or based on platform) this results in very different blur appearance.

![Screenshot 2023-01-18 at 12 14 45](https://user-images.githubusercontent.com/17763524/213158213-223fe4e5-48b4-4df5-816c-e73a34f3504e.jpg)
![Screenshot 2023-01-18 at 12 15 20](https://user-images.githubusercontent.com/17763524/213158223-482ed795-89ce-49d5-b18f-322a9d65e122.jpg)

Maybe this is a bug (in 4.0.b12)? 
But I suspect not, and that it's more a limitation in how shadow blurring works. So, what I'd like to propose or discuss is some way of defining blur in a resolution independent relative way so that the relative shadow blur becomes the same regardless of shadow map resolution. Perhaps an alternative "relative blur size" parameter on lights that is automatically adapted to the shadow map size?

**Comment by Calinou** [comment]
This is done by design. People generally expect shadows to become sharper as you increase shadow resolution, even if this is physically incorrect. It's generally what you see in most games when you increase the shadow quality setting :slightly_smiling_face: 

If https://github.com/godotengine/godot-proposals/issues/5728 is implemented, you'll be able to recreate the mechanism you want with a few lines of code. In the meantime, you could loop over all lights when loading a level and multiply the blur radius according to the currently selected shadow resolution.

**Reply by lostminds** [reply]
This was a surprising reply to me. If you use "sharp shadows", as in with little or no blur, I guess you could expect higher resolution shadow maps to produce even "sharper" shadows. But I think what you'd then mean would actually just be more detailed shadows.

If we have a game with very soft shadows with a lot of blur as a design choice I think it is more unexpected that this design will change with the shadow map resolution. Basically I'd consider the shadow map resolution a performance/quality setting that improves fidelity while blur size to me is more of a design parameter (like color or light intensity) that should as far as possible produce similar output regardless of performance/quality settings. But perhaps that's an indication of a wider discussion regarding shadows, since there are so many different parameters to tweak that all affect the appearance, and it's hard to tell sometimes what are core "design properties" and what parameters are "quality settings" since these more technical quality and performance settings often have such a lot of impact on the appearance of the shadows.
